### ISSUE TITLE
env GOARCH=js gopherjs test fails

### ISSUE THREAD

**Issue by bep** [gopherjs/gopherjs#594]
I created a simple CLI wrapper to switch between go and gopherjs development in LiteIDE:

https://github.com/bep/go-or-gopherjs

I thought I'd be [clever](https://github.com/bep/go-or-gopherjs/blob/master/main.go#L22) and use `GOARCH=js` for this, but that fails on my `macOS` computer.

```
▶ env GOARCH=js gopherjs test
../../../../../../../../usr/local/go/src/syscall/syscall_unix.go:196:38: undeclared name: _Socklen
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:16:28: undeclared name: BpfInsn
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:21:36: undeclared name: BpfInsn
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:82:9: undeclared name: IFNAMSIZ
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:108:27: undeclared name: Timeval
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:118:32: undeclared name: Timeval
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:127:25: undeclared name: BpfStat
../../../../../../../../usr/local/go/src/syscall/bpf_bsd.go:146:25: undeclared name: BpfInsn
../../../../../../../../usr/local/go/src/syscall/flock.go:13:30: undeclared name: SYS_FCNTL
../../../../../../../../usr/local/go/src/syscall/flock.go:16:42: undeclared name: Flock_t
../../../../../../../../usr/local/go/src/syscall/flock.go:16:42: too many errors
```

It is not a big deal for my use case, but I thought I should give you a heads up.

**Comment by shurcooL** [comment]
Thanks for reporting.

The `gopherjs` tool does not use the `GOARCH` environment variable in any way. Internally, GopherJS always uses `js` as `GOARCH` value when compiling user code, but it does other internal stuff when compiling standard library.

The fix is to make the `gopherjs` tool really ignore `GOARCH` env var. In other words, setting it to any value should have no effect; it shouldn't cause this failure.

**Comment by shurcooL** [comment]
We could also consider making it use the value of `GOARCH`, and error out if it's set to anything other than `js`. Since `gopherjs` can't compile for any other architectures.

That might be better since it's more consistent with the `cmd/go` tool.

**Comment by shurcooL** [comment]
We've originally discussed two possible fixes in this issue:

> The `gopherjs` tool does not use the `GOARCH` environment variable in any way. Internally, GopherJS always uses `js` as `GOARCH` value when compiling user code, but it does other internal stuff when compiling standard library.
>
> 1. One fix is to make the `gopherjs` tool really ignore `GOARCH` env var. In other words, setting it to any value should have no effect; it shouldn't cause this failure.
>
> 2. We could also consider making it use the value of `GOARCH`, and error out if it's set to anything other than `js` (which is the default and the only supported value). Since `gopherjs` can't compile for any other architectures.

Upon further consideration, after seeing issues #601 and gopherjs/gopherjs.github.io#66, we've decided (see discussion in https://github.com/gopherjs/gopherjs.github.io/pull/66#issuecomment-286519322) to revisit this issue and change gopherjs to completely ignore `GOARCH` environment variable, instead of requiring it to be unset or equal to `js`.

I'll make a PR for that shortly.

**Comment by shurcooL** [comment]
Actually, this is the wrong issue I've reopened to track this.

I'll close this because this issue continues to be resolved by 69499efe08d29030cc47842bf2a214ca2de4edc5.
